src/main/java/io/cdap/plugin/google/sheets/source/SheetTransformer.java

```diff
@@ -69,8 +69,9 @@ public static StructuredRecord transform(RowRecord rowRecord, Schema schema, boo
         builder.set(metadataRecordName, rowRecord.getMetadata());
       } else {
         ComplexSingleValueColumn complexSingleValueColumn = rowRecord.getHeaderedCells().get(name);
-        if (complexSingleValueColumn == null || complexSingleValueColumn.getData() == null
-            || complexSingleValueColumn.getSubColumns() == null || complexSingleValueColumn.getSubColumns().isEmpty()) {
+        if (complexSingleValueColumn == null ||
+          (!field.getSchema().getNonNullable().getType()
+            .equals(Schema.Type.RECORD) && complexSingleValueColumn.getData() == null)) {
```

Comment:
why remove `getSubColumns` check?

Reply:
In case of normal fields:
get data will have Value and getSubColumns will be empty; Handled here.

In case of record fields:
get data will be null and getSubColumns will have data: Handled in Record specific code.

In case of cx, they have a wrong use case where they don't want pipeline to fail:
type is record but they set schema to String, and get data is null for that, so we are setting it to null.